state/factory/workingset.go

```diff
@@ -142,34 +143,39 @@ func withActionCtx(ctx context.Context, selp action.SealedEnvelope) (context.Con
 
 func (ws *workingSet) runAction(
 	ctx context.Context,
-	elp action.SealedEnvelope,
+	selp action.SealedEnvelope,
 ) (*action.Receipt, error) {
 	if protocol.MustGetBlockCtx(ctx).GasLimit < protocol.MustGetActionCtx(ctx).IntrinsicGas {
 		return nil, action.ErrGasLimit
 	}
 	// Reject execution of chainID not equal the node's chainID
-	if !action.IsSystemAction(elp) {
-		if err := validateChainID(ctx, elp.ChainID()); err != nil {
+	if !action.IsSystemAction(selp) {
+		if err := validateChainID(ctx, selp.ChainID()); err != nil {
 			return nil, err
 		}
 	}
+	// for replay tx, check against deployer whitelist
+	chainCtx := protocol.MustGetBlockchainCtx(ctx)
+	if selp.Encoding() == uint32(iotextypes.Encoding_ETHEREUM_UNPROTECTED) && !chainCtx.IsDeployerWhitelisted(selp.SrcPubkey().Address()) {
```

Comment:
May not be required to check in workingset if it has already been rejected in api, since it has no chance to get here

Reply:
If it's not required here, the whitelist could only just introduced in api scope

Reply:
for safety we need it here, b/c someone can run their own API node w/o the whitelist check.
similar to validate chainID, it is also rejected in API, but also need to check in workingset